tests/cases/fourslash/codeFixImportNonExportedMember8.ts

```diff
@@ -0,0 +1,22 @@
+/// <reference path="fourslash.ts" />
+
+// @module: esnext
+// @filename: /a.ts
+////const a = 1;
+////type T = number;
+////export { a };
+
+// @filename: /b.ts
+////import { T } from "./a";
+
+goTo.file("/b.ts");
+verify.codeFix({
+    description: [ts.Diagnostics.Export_0_from_module_1.message, "T", "./a"],
+    index: 0,
+    newFileContent: {
+        "/a.ts":
+`const a = 1;
+type T = number;
+export { a, type T };`,

```

Comment:
What compiler option can I use to do this? Or the type `T,` in this case, should be exported like this
```ts
export { a, T };`,
```

Reply:
`importsNotUsedAsValues` and/or `preserveValueImports` - but I would hold off on changes until @andrewbranch can take a look.

Reply:
Just `isolatedModules` I think:

<img width="727" alt="image" src="https://user-images.githubusercontent.com/3277153/195684899-d5c972d6-9c02-481e-9781-9029f7273ee3.png">

https://www.typescriptlang.org/play?isolatedModules=true#code/PTAEAEEsGcHsBsCGAXApgEwLK3QV3qtAFyjIBOuqAUMgJ4AOqoAKqALygDeAvgNxWoAHvVhlkXFqD5UgA

Reply:
I think the current implementation handles this without checking `isolatedModules`, it works for _type declarations_ with the following algorithm

1. If the file contains `export type { }`, then QF adds an `export type { T }` type declaration to it
2. If the file contains `export {}` then QF adds the export with type specifier `export { type T }`
3. Otherwise, QF adds the export modifier - `export type T = {}`;

Is it ok? Or is this the wrong way?

Reply:
I like that overall, but would modify (2) to only include `type` if `isolatedModules` is set, or if the export declaration already uses `type` modifiers within.

Reply:
@andrewbranch Oke.